### ISSUE TITLE
Race condition in registerRoute

### ISSUE THREAD

**Issue by WithoutPants** [stashapp/stash#5124]
Using the example `react-component` plugin, the registered route `/plugin/test-react` does not reliably work if the URL is entered or refreshed. I believe this is occurring because the `PluginApi.register.route` is being called after `PluginRoutes` is first rendered, and registering the route does not trigger a re-render. An alternative implementation for the register route functionality will need to be devised which will respond to new registrations after the initial render.

**Comment by angrytoenail** [comment]
I recently bumped into this issue after following the docs and testing the example react-component plugin. Navigating to `/plugin/test-react` would work as expected when clicking a link from within Stash, but refreshing or opening a new tab would result in a 404 error.

@WithoutPants you mentioned you thought the cause might be because of a race condition on the frontend, but  it seems to me the more likely culprit is the backend server... it's the backend that's ultimately responsible for routing requests (at least until the frontend loads and React Router takes over). There's a bunch of [routes](https://github.com/stashapp/stash/blob/dcfb3b7d37f2232e165dc660adb8599ebad867c2/internal/api/server.go#L221) defined under the `/plugin` prefix (`/plugin/<plugin-id>/javascript/*.js`, `/plugin/<plugin-id>/css/*.css`, `/plugin/<plugin-id/assets/*`, etc), but crucially the server logic returns a `http.NotFound` if it can't find anything to serve. This is what's happening when navigating to `/plugin/test-react`...

Realising this, I tested registering a route directly on the root URL (e.g. `http://stash:9999/my-plugin`) and it works exactly as expected, including refreshing the page, linking to it from within Stash and navigating to it directly from a new window/tab.
